### ISSUE TITLE
libglvnd support

### ISSUE THREAD

**Issue by matwey** [ptitSeb/gl4es#125]
Hello.

Are there any plans to support libglvnd? It is becoming a standard way to proxy gl library in major Linux distros.

**Comment by ptitSeb** [comment]
First time I heard of this lib. I need to check what it means to support libglvnd on gl4es side.

**Comment by matwey** [comment]
It is a library for peaceful coexistence of multiple OpenGL implementations on the same system.

**Comment by ptitSeb** [comment]
Yes, I have read, but I'm unsure what I have to do in gl4es to be compatible with it.

**Comment by ptitSeb** [comment]
Mmm, it seems I need to look in `include/libglxabi.h` and implement what is written to have support.
I'll see what I can do later about that.

**Comment by rapgro** [comment]
Well, we'd need this support for an official package in Fedora, otherwise there can't be any official approval due to proprietary dependencies.

**Comment by ptitSeb** [comment]
@rapgro do you have a precise list of extension / functions I need to support libglvnd?

**Comment by rapgro** [comment]
Not really. Just curious because in the downstream review someone recommends to use libglvnd for box64.

**Comment by matwey** [comment]
https://www.khronos.org/registry/OpenGL/extensions/EXT/GLX_EXT_libglvnd.txt

**Comment by matwey** [comment]
https://github.com/NVIDIA/libglvnd/blob/master/src/EGL/icd_enumeration.md

**Comment by kwizart** [comment]
> Not really. Just curious because in the downstream review someone recommends to use libglvnd for box86.

More Accurately, I recommends that for FLOSS drivers on armhfp and aarch64, box86 somehow default to expect plain GL drivers (from Mesa) over GLES ones.

**Comment by fredldotme** [comment]
I have started some work in this area a year ago with regards to making it run X11 applications using GLX transparently on [libhybris](https://github.com/libhybris/libhybris) devices like most on Ubuntu Touch.

https://github.com/fredldotme/gl4es/commits/box64andwine

This should, in the long run, allow loading libGL through libglvnd and translate those desktop GL functions and GLX use into something Android EGL can use on a GNU/Linux mobile OS.

I'd like to know what and how much of interest it is, so that I can prepare and clean up a PR or two.

**Comment by ptitSeb** [comment]
I'm sure this can interest some people yes! That's some interresting stuffs. Need to put that behind a build flag like "HYBRIS" maybe?

**Comment by fredldotme** [comment]
GLVND and HYBRIS would be separate build options but yeah! This would allow us to land GL4ES in the mobile operating system I help maintain, possibly, potentially (not making any promises right now ;) )
